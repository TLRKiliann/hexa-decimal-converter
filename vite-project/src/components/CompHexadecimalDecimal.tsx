import React, { useState } from 'react'
import CompLabelInputHexadecimal from './subcomponents/CompHexaDecimalDecimal/CompLabelInputHexadecimal.tsx'
import CompResultHexadecimal from './subcomponents/CompHexaDecimalDecimal/CompResultHexadecimal.tsx'

export default function CompHexadecimalDecimal() {
    const [hexadecimal, setHexadecimal] = useState<string>("");
    const [resultHexa, setResultHexa] = useState<number | undefined>();
  
    const converterTarget = (event: React.ChangeEvent<HTMLInputElement>) => {
      const etv = event?.target.value;
      setHexadecimal(etv);
    }
  
    const convertResult = () => {
      const hexaConv = parseInt(hexadecimal,16);
      console.log(hexaConv, "hexaConv");
      setResultHexa(hexaConv);
      setHexadecimal("");
    }

    return (
        <div>

            <div className="labelinput--box">
                <CompLabelInputHexadecimal
                    hexadecimal={hexadecimal}
                    converterTarget={(event)=>converterTarget(event)}
                />
            </div>

            <button type="button" onClick={convertResult}>Convert</button>
            
            <div className="box--binary--result">
                <CompResultHexadecimal 
                    resultHexa={resultHexa}
                />
            </div>

        </div>
    )
}
