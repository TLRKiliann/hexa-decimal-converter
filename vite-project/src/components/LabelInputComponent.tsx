import React from 'react'

type InputLblProps = {
    value: string | undefined;
    nameValue: string;
    converterTarget: (event: React.ChangeEvent<HTMLInputElement>) => void;
    children: string | JSX.Element | JSX.Element[];
}

export default function LabelInputComponent({value, nameValue, converterTarget, children}: InputLblProps) {
    return (
        <>
            <label>{children} :</label>
            <input
                data-testid="testid-input"
                type="text"
                value={value}
                onChange={(event)=>converterTarget(event)}
                placeholder={nameValue}
            />
        </>
    )
}
