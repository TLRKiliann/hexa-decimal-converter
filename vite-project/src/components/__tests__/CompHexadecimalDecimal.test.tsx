import { test, expect, describe, vi } from "vitest"
import { fireEvent, screen, render } from '@testing-library/react'
import { create } from "react-test-renderer"
import "@testing-library/jest-dom"
import CompHexadecimalDecimal from '../CompHexadecimalDecimal';
import LabelInputComponent from "../LabelInputComponent"

test('Snapshat of CompHexadecimalDecimal', () => {
    const tree = create(<CompHexadecimalDecimal />)
    expect(tree.toJSON()).toMatchSnapshot()
});

describe("testing children FirstComp", () => {
  test('testing function handleReverse', () => {
    const beverage = "tested";
    const convertResult = vi.fn(beverage => beverage)
    convertResult(beverage)
    render(
      <LabelInputComponent value={undefined} nameValue={""} converterTarget={function (): void {
        throw new Error("Function not implemented.")
      }}>
        <button data-testid="testid-btn">
          Convert
        </button>
      </LabelInputComponent>
    )
    fireEvent.click(screen.getByTestId('testid-btn'))
    //expect(convertResult).toHaveReturnedWith("tested")
    expect(convertResult).toHaveBeenCalledTimes(1)
    expect(convertResult).toBeDefined();
  })
});