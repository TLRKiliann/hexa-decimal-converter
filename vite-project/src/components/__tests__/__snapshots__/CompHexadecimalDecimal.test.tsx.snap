// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of CompHexadecimalDecimal 1`] = `
[
  <div
    className="labelinput--box"
  >
    <label>
      Hexadecimal
       :
    </label>
    <input
      data-testid="testid-input"
      onChange={[Function]}
      placeholder="Hexadecimal"
      type="text"
      value=""
    />
  </div>,
  <button
    data-testid="testid-btn"
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>,
  <div
    className="box--result"
  >
    <p>
      Result : 
      Not binary !!!
    </p>
  </div>,
]
`;
